kore/src/Kore/Parser/Lexeme.hs

```diff
@@ -97,35 +100,37 @@ sourcePosToFileLocation
     , column   = unPos column'
     }
 
-{-|'idParser' parses either an @object-identifier@, or a
-@set-variable-identifier@.
-
-The @set-variable-@ version always starts with @\@@, while the @object-@
-one does not.
+{-| Generic identifier parser. Takes a parser for the string of the identifier
+ and returns an 'Id' annotated with position.
 -}
-idParser :: Parser Id
-idParser = do
+genericIdParser :: Parser String -> Parser Id
+genericIdParser idRawParser = do
     pos <- sourcePosToFileLocation <$> getSourcePos
-    name <- lexeme (objectIdRawParser KeywordsForbidden <|> setVarIdRawParser)
+    name <- lexeme idRawParser
     return Id
         { getId = Text.pack name
         , idLocation = AstLocationFile pos
         }
 
+variableIdRawParser :: Parser String
+variableIdRawParser = elemVarIdRawParser <|> setVarIdRawParser
+
+{-|'variableIdParser' parses either an @element-variable-identifier@, or a
```

Comment:
The documentation here and below should be updated to reflect the new syntax document.